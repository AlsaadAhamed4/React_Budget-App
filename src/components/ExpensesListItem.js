import React from 'react';
import { removeExpenseAction, editExpenseAction } from '../Actions/ExpenseAction';
import { NavLink } from 'react-router-dom';

const ExpensesListItem = ({ description, amount, createdAt, id, dispatch }) => (
    <div>
        <NavLink to={`/edit/${id}`}><h1>{description}</h1></NavLink>
        <p>{amount}</p>
        <p>{createdAt}</p>
    </div >
);

export default ExpensesListItem;