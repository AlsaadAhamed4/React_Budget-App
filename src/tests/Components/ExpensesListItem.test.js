import React from 'react';
import { shallow } from 'enzyme';
import ExpensesListItem from '../../components/ExpensesListItem';

//testing data
const expenses = [
    {
        id: 'aasd',
        description: 'hola',
        amount: 1000,
        createdAt: 1200
    },
    {
        id: 'wwww',
        description: 'Hola Test 1',
        amount: 1100,
        createdAt: 1100
    },
    {
        id: 'eeee',
        description: 'Hola Test 2 ',
        amount: 1200,
        createdAt: 1200
    }
];

test('should render expensesListItem', () => {
    const wrapper = shallow(<ExpensesListItem {...expenses[1]} />);
    expect(wrapper).toMatchSnapshot();
})