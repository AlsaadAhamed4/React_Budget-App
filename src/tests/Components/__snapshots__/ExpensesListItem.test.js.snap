// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expensesListItem 1`] = `
<div
  className="expenselist-item"
>
  <[object Object]
    className="expenselist-item--link"
    to="/edit/wwww"
  >
    <div
      className="expenselist-item__name"
    >
      <h1>
        Hola Test 1
      </h1>
      <p>
        January 1st, 1970
      </p>
    </div>
    <div
      className="expenselist-item__amount"
    >
      <p>
        $11.00
      </p>
    </div>
  </[object Object]>
</div>
`;
